src/Brush.elm

```diff
@@ -0,0 +1,1102 @@
+
+
+{-| By default the brush will use the meta/alt and shift keys to change behavior. You can disable this with this function.
+-}
+keyboardModifiersEnabled : Bool -> Brush dimension -> Brush dimension
+keyboardModifiersEnabled enabled (Brush model) =
+    Brush { model | keysEnabled = enabled }
+
+
+{-| Exposes the selection for a single dimensional brush, where the first number should always be less than the second.
+-}
+selection1d : Brush OneDimensional -> Maybe ( Float, Float )
+selection1d (Brush { selection, x }) =
+    Maybe.map
+        (\{ top, left, bottom, right } ->
+            if x then
+                ( min right left, max right left )
+
+            else
+                ( min top bottom, max top bottom )
+        )
+        selection
+
+
+{-| Exposes the selection for a two dimensional brush.
+-}
+selection2d : Brush TwoDimensional -> Maybe Extent
+selection2d (Brush { selection }) =
+    selection
+
+
+{-| -}
+type TransitionOption
+    = Instantly
+
+
+{-| Perfom the update to the brush instantly, rather than with an animation (animations are not supported yet.)
+-}
+instantly : TransitionOption
+instantly =
+    Instantly
+
+
+{-| Programatically set the selection of the Brush.
+-}
+setSelection1d : TransitionOption -> ( Float, Float ) -> Brush OneDimensional -> Brush OneDimensional
+setSelection1d _ ( a, b ) (Brush model) =
+    if model.x then
+        Brush
+            { model
+                | selection =
+                    Just
+                        { left = clamp model.extent.left a b
+                        , right = clamp a b model.extent.right
+                        , top = model.extent.top
+                        , bottom = model.extent.bottom
+                        }
+            }
+
+    else
+        Brush
+            { model
+                | selection =
+                    Just
+                        { left = model.extent.left
+                        , right = model.extent.right
+                        , top = clamp model.extent.top a b
+                        , bottom = clamp a b model.extent.bottom
+                        }
+            }
+
+
+{-| Programatically set the selection of the Brush (in two dimensions).
+-}
+setSelection2d : TransitionOption -> Extent -> Brush TwoDimensional -> Brush TwoDimensional
+setSelection2d _ sel (Brush model) =
+    Brush
+        { model
+            | selection =
+                Just
+                    { left = max sel.left model.extent.left
+                    , right = min sel.left model.extent.left
+                    , top = max model.extent.top sel.top
+                    , bottom = min model.extent.bottom sel.bottom
+                    }
+        }
+
+
+{-| Clears the selection programmatically.
+
+     brush
+       |> Brush.clearSelection
+       |> Brush.selection1d --> Nothing
+
+-}
+clearSelection : Brush dim -> Brush dim
+clearSelection (Brush model) =
+    Brush { model | selection = Nothing }
```

Comment:
Does `clearSelection` also get called implicitly when you click outside the brush selection? I didn't see this command in the timeseries example.

Reply:
Yes, this is intended for programmatic use. 

When using the built in event handling, if you click, but don't drag we will clear the selection rather than have a zero width selection.